runtime/lua/vscode-neovim.lua

```diff
@@ -1,30 +1,28 @@
 local api = require("vscode-neovim.api")
-local defaults = require("vscode-neovim.defaults")
-local cursor = require("vscode-neovim.cursor")
-
-require("vscode-neovim.highlight")
-require("vscode-neovim.filetype")
 
-local M = {}
+local default_optons = require("vscode-neovim.default-options")
+local cursor = require("vscode-neovim.cursor")
+local highlight = require("vscode-neovim.highlight")
+local filetype = require("vscode-neovim.filetype")
+local numbers = require("vscode-neovim.numbers")
+local autocmds = require("vscode-neovim.autocmds")
 
-M.notify = api.notify
-M.call = api.call
-M.call_range = api.call_range
-M.notify_range = api.notify_range
-M.call_range_pos = api.call_range_pos
-M.notify_range_pos = api.notify_range_pos
+default_optons.setup()
+cursor.setup()
+highlight.setup()
+filetype.setup()
+numbers.setup()
+autocmds.setup()
 
-M.setup = function()
-  defaults.setup()
-  cursor.setup()
-end
+local vscode = {
+  action = api.action,
+  call = api.call,
+  on = api.on,
+  has_config = api.has_config,
+  get_config = api.get_config,
+  update_config = api.update_config,
+}
 
-vim.api.nvim_create_autocmd("InsertLeave", {
-  callback = function()
-    vim.fn.VSCodeNotify("hideSuggestWidget")
-    vim.fn.VSCodeNotify("closeParameterHints")
-    vim.fn.VSCodeNotify("editor.action.inlineSuggest.hide")
-  end,
-})
+_G._vscode = vscode
```

Comment:
I thought we agreed this is an anti-pattern :p

Reply:
Easy to test, I will delete it.